### PR TITLE
Hardening tests for v3 sources.

### PR DESCRIPTION
**Pull request by pranavkm**
[Review on Reviewable](https://reviewable.io/reviews/nuget/nuget3/28)

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @emgarten

File: test/NuGet.Protocol.Core.v3.Tests/ServiceIndexResourceV3ProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            // Assert
+            Assert.False(result.Item1);
+            Assert.Null(result.Item2);
+        }
+
+        [Fact]
+        public async Task TryCreate_ReturnsFalse_IfSourceLocationReturnsFailureCode()
+        {
+            // Arrange
+            var source = "https://fake.server/does-not-exist.json";
```
Comment (yishaigalatzer) [inline test/NuGet.Protocol.Core.v3.Tests/ServiceIndexResourceV3ProviderTests.cs:60]: nit: don't use fake (as its a related product name)

**Comment by yishaigalatzer** [comment]
:shipit: when @emgarten is happy

**Comment by emgarten** [comment]
These are good tests.

:shipit:

**Comment by yishaigalatzer** [comment]
@pranavkm can this be merged?

**Pull request closed without merging**